lightning/src/ln/channelmanager.rs

```diff
@@ -4658,6 +4658,14 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 							msg,
 						});
 					}
+
+					let mut pending_events = self.pending_events.lock().unwrap();
```

Comment:
Also not 100% sure whether I covered all the cases where we should emit this event. Two feels like I'm missing something. Will have a further look.